docs/user-guide/content-type.md

```diff
@@ -0,0 +1,232 @@
+cover any of the specific use cases.
+This protocol can be thought of a wider _"lower level"_ spec, which only
+defines what fields should a payload have.
+
+To account for this gap, MLServer introduces support for **content types**,
+which offer a way to let MLServer know how it should _"decode"_ V2-compatible
+payloads.
+When shaped in the right way, these payloads should _"encode"_ all the
+information required to extract the higher level Python type that will be
+required for a model.
+
+To illustrate the above, we can think of a Scikit-Learn pipeline, which takes
+in a Pandas DataFrame and returns a NumPy Array.
+Without the use of **content types**, the V2 payload itself would probably lack
+information about how should this payload be treated by MLServer.
+Likewise, the Scikit-Learn pipeline wouldn't know how to treat a raw V2
+payload.
+In this scenario, the use of content types allows us to specify information on
+what's the actual _"higher level"_ information encoded within the V2 protocol
+payloads.
+
+![Content Types](../assets/content-type.svg)
+
+## Usage
+
+To let MLServer know that a particular payload must be decoded / encoded as a
+different Python data type (e.g. NumPy Array, Pandas DataFrame, etc.), you can
+specifity it through the `content_type` field of the `parameters` section of
+your request.
+
+As an example, we can consider the following dataframe, containing two columns:
+Age and First Name.
+
+| First Name | Age |
+| ---------- | --- |
+| Joanne     | 34  |
+| Michael    | 22  |
+
+This table, could be specified in the V2 protocol as the following payload, where we declare that:
+
+- The whole set of inputs should be decoded as a Pandas Dataframe (i.e. setting
+  the content type as `pd`).
+- The Age column should be decoded as a UTF-8 string (i.e. setting the content
+  type as `str`).
+
+```{code-block} json
+---
+emphasize-lines: 2-4, 9-11
+---
+{
+  "parameters": {
+    "content_type": "pd"
+  },
+  "inputs": [
+    {
+      "name": "First Name",
+      "datatype": "BYTES",
+      "parameters": {
+        "content_type": "str"
+      },
+      "shape": [2],
+      "data": ["Joanne", "Michael"]
+    },
+    {
+      "name": "Age",
+      "datatype": "INT32",
+      "shape": [2],
+      "data": [34, 22]
+    },
+  ]
+}
+```
+
+To learn more about the available content types and how to use them, you can
+see all the available ones in the [Available Content
+Types](#available-content-types) section below.
+For a full end-to-end example on how content types work under the hood, you can
+also check out this [Content Type Decoding
+example](../examples/content-type/README.md).
+
+```{note}
+It's important to keep in mind that content types can be specified at both the
+**request level** and the **input level**.
+The former will apply to the **entire set of inputs**, whereas the latter will
+only apply to a **particular input** of the payload.
+```
+
+## Available Content Types
```

Comment:
should we have a note about the default `content_type` if it is not specified by the user?

Reply:
The default content type is dependant on each runtime, but maybe we can just mention that instead of listing all the defaults?

Reply:
a reference to where uses can get this information is fine as it seems it might affect how they construct the payload?

Reply:
Done